core/domain/user_services.py

```diff
@@ -931,24 +1021,175 @@ def create_new_user(gae_id, email):
 
+def create_new_profiles(gae_id, email, modifiable_user_data_list):
+    """Creates new profiles for the users specified in the
+    modifiable_user_data_list and commits them to the datastore.
+
+    Args:
+        gae_id: str. The GAE ID of the full (parent) user trying to create new
+            profiles.
+        email: str. The email address of the full (parent) user trying to create
+            new profiles.
+        modifiable_user_data_list: list(ModifiableUserDataV1). The list of
+            modifiable user data objects used for creation of new profiles.
+
+    Returns:
+        list(UserSettings). List of UserSettings objects created for the new
+        users.
+
+    Raises:
+        Exception. The pin for parent user trying to create a new profile
+            must be set.
+        Exception. The user_id is already set for any user in its corresponding
+            modifiable_user_data object.
+    """
+
+    def _create_new_profile_transactional(
+            user_settings, user_auth_details):
+        """Save user models for new users as a transaction.
+
+        Args:
+            user_settings: UserSettings. The user settings domain object
+                corresponding to the newly created user.
+            user_auth_details: UserAuthDetails. The user auth details domain
+                object corresponding to the newly created list of users.
+        """
+        _save_user_auth_details(user_auth_details)
+        _save_user_settings(user_settings)
+
+    # As new profile user creation is done by a full (parent) user only.
+    parent_user_auth_details = get_auth_details_by_gae_id(gae_id, strict=True)
+    if parent_user_auth_details.pin is None:
+        raise Exception(
+            'Pin must be set for a full user before creating a profile.')
+    parent_user_id = parent_user_auth_details.user_id
+    user_settings_list = []
+    for modifiable_user_data in modifiable_user_data_list:
+        if modifiable_user_data.user_id:
+            raise Exception('User id cannot already exist for a new user.')
+        user_id = user_models.UserSettingsModel.get_new_id('')
+        user_settings = UserSettings(
+            user_id, gae_id, email, feconf.ROLE_ID_LEARNER,
+            preferred_language_codes=[constants.DEFAULT_LANGUAGE_CODE]
+        )
+        user_settings.populate_from_user_data(modifiable_user_data)
+
+        user_auth_details = UserAuthDetails(
+            user_id, None, modifiable_user_data.pin, parent_user_id)
+
+        transaction_services.run_in_transaction(
+            _create_new_profile_transactional, user_settings,
+            user_auth_details
+        )
+        user_settings_list.append(user_settings)
+    return user_settings_list
+
+
+def update_multiple_users_data(modifiable_user_data_list):
+    """Updates user settings and user auth model details for the users
+    specified in the modifiable_user_data_list.
+
+    Args: modifiable_user_data_list: list(ModifiableUserDataV1). The list of
```

Comment:
Wrong docstring format, please fix. I commented on this before.

Reply:
Sorry, fixed now.